### PR TITLE
Add Bitbucket doc

### PR DESCRIPTION
**Pull request by harrietgrace**
Docs for getting started with Buildkite. This will live under "How Tos" > "Connecting to source control". 

Pretty much just what used to be displayed in the UI when you added a pipeline with a bitbucket URL. Added some little intro bits, and will be replacing the screenshots with newer ones, once I'm on a retina screen rather than my little laptop 😄

- [x] Add a sidebar item for this to the docs 
- [x] Add screenshots
- [x] Update the doc that is shown in the UI for bitbucket pipelines
- [ ] Reviews
- [ ] Do the thing 🚀

### CODE REVIEW

**Comment by keithpitt** [comment]
Words look good to me! This doc will be a great place for us to link to in the UI/support and for third-parties that want to tell customers how to connect BK to Bitbucket.

File: pages/how_tos/bitbucket.md.erb [hunk 1]
```diff
@@ -1,27 +1,17 @@
 # Bitbucket Repo Access
 
-- Bitbucket's build status API
-- creating or updating a PR triggers Buildkite to create a new build
-- after the build completes, the status icon in Bitbucket will update
+Connecting to Bitbucket's build status API allows repository events to trigger Buildkite builds. Events such as creating or updating a PR triggers the creation of a new build in the associated Buildkite pipeline. After the build completes in Buildkite, the status icon in Bitbucket will update to reflect the outcome.
```
Comment (ticky) [inline pages/how_tos/bitbucket.md.erb]: > Events such as creating or updating a PR triggers the creation

Sounds like it should be

> Events such as creating or updating a PR trigger the creation

File: pages/how_tos/bitbucket.md.erb [hunk 2]
```diff
@@ -30,8 +20,20 @@
 	* Merged
 	* Declined 
 
-Once you've entered all of this information, the form should look like this:
-<screenshot here>
+Once you've filled it out, the form should look like this:
+<%= image "webhook-form.png", width: 930/2, height: 1578/2, alt: "Screenshot of a completed Bitbucket webhook form" %>
+
+Click the `save` button, and the webhook will be created. Set up a Buildkite pipeline using this repository's url, and the repository events that you selected will trigger the creation of Buildkite builds 🎉
```
Comment (ticky) [inline pages/how_tos/bitbucket.md.erb]: `save` -> `Save`, to match the image

File: pages/how_tos/bitbucket.md.erb [hunk 3]
```diff
@@ -30,8 +20,20 @@
...
+
+As you set up a new Buildkite pipeline with a Bitbucket repository, we'll display these instructions again, populated with your unique account details.
```
Comment (ticky) [inline pages/how_tos/bitbucket.md.erb]: “as” feels awkward as the start of this sentence - “when” or “if” seem to sit better

File: pages/how_tos/bitbucket.md.erb [hunk 4]
```diff
@@ -30,8 +20,20 @@
 	* Merged
 	* Declined
```
Comment (sj26) [inline pages/how_tos/bitbucket.md.erb]: I can see people skimming and skipping the repository event.

Maybe we could break this into two sub-lists, like:

1. From the list of triggers, select:  
   - Repository:  
     * Push
   - Pull Request:  
     * Created
     * Updated
     * Approved
     * Approval Removed
     * Merged
     * Declined

File: pages/how_tos/bitbucket.md.erb (hunk 2 above)
Comment (sj26) [inline pages/how_tos/bitbucket.md.erb]: Should this `` `save` `` look like code? I try to use emphasis for text from a UI, like:

> Click the _Save_ button, and...

File: pages/how_tos/bitbucket.md.erb (hunk 3 above)
Comment (sj26) [inline pages/how_tos/bitbucket.md.erb]: "populated with your unique account details" sounds a little like the URL might be the same for the account, but it changes per pipeline. Or maybe I'm overthinking!

File: pages/how_tos/bitbucket.md.erb [hunk 5]
```diff
@@ -30,8 +20,20 @@
...
+Once you've filled it out, the form should look like this:
+<%= image "webhook-form.png", width: 930/2, height: 1578/2, alt: "Screenshot of a completed Bitbucket webhook form" %>
+
+Click the `save` button, and the webhook will be created. Set up a Buildkite pipeline using this repository's url, and the repository events that you selected will trigger the creation of Buildkite builds 🎉
+
+As you set up a new Buildkite pipeline with a Bitbucket repository, we'll display these instructions again, populated with your unique account details. 
+
+## Connecting Buildkite to Bitbucket
+
+Connecting Bitbucket as an app in Buildkite will allow webhooks from Bitbucket access to your Buildkite account. Enabling this will allow Bitbucket's build status icons to reflect the state of your Buildkite builds. This step is optional - not adding Bitbucket as a connected app will not affect your ability to trigger Buildkite builds from your Bitbucket repository.
```
Comment (sj26) [inline pages/how_tos/bitbucket.md.erb]: > Connecting Bitbucket as an app in Buildkite will allow webhooks from Bitbucket access to your Buildkite account.

Should this maybe be the other way around? 

> Connecting Bitbucket as an app in Buildkite will allow Buildkite limited access to your Bitbucket account.

**Comment by sj26** [comment]
This is super excellent. We've been needing source control guides for aaaaages. :+1:

File: pages/how_tos/bitbucket.md.erb (hunk 5 above)
Comment (harrietgrace) [inline pages/how_tos/bitbucket.md.erb]: Riiight, so Buildkite has no access to your bitbucket before you add it as an app? And the webhooks from bitbucket have access to your Buildkite without adding it as an app.

Comment (sj26) [inline pages/how_tos/bitbucket.md.erb]: image

Comment (sj26) [inline pages/how_tos/bitbucket.md.erb]: The special generated webhook URL includes a token which authenticates what Bitbucket sends to Buildkite without another "connect to Buildkite" step. 👮

**Comment by harrietgrace** [comment]
@ticky @sj26 Many thanks! Updated with your suggestions 🙌🏻

File: pages/how_tos/bitbucket.md.erb [hunk 6]
```diff
@@ -1,37 +1,44 @@
 # Bitbucket Repo Access
```
Comment (keithpitt) [inline pages/how_tos/bitbucket.md.erb]: The other "How To" titles are just "GitLab" and "GitLab Enterprise". Does that mean this one should just be "Bitbucket"?

File: pages/how_tos/bitbucket.md.erb [hunk 7]
```diff
@@ -1,37 +1,44 @@
...
 
-- Bitbucket's build status API
-- creating or updating a PR triggers Buildkite to create a new build
-- after the build completes, the status icon in Bitbucket will update
+Connecting to Bitbucket's build status API allows repository events to trigger Buildkite builds. Events such as creating or updating a PR trigger the creation of a new build in the associated Buildkite pipeline. After the build completes in Buildkite, the status icon in Bitbucket will update to reflect the outcome.
```
Comment (keithpitt) [inline pages/how_tos/bitbucket.md.erb]: I read this as a "updating a PR trigger" as in "what is a _pull request trigger_"

**Comment by keithpitt** [comment]
per-step
single-status

Here are some images you can use if you want!

**Comment by harrietgrace** [comment]
@toolmantim want to check this out before it's merged?

**Comment by toolmantim** [comment]
@harrietgrace yep, I will review now!

**Comment by toolmantim** [comment]
@harrietgrace I've done a quick update of this so we can send it live and 🚀, including a bunch of new words and images. I'll merge this in and send it live so we can check it out in production. If I could get you to check for typos and grammar, then I'm keen to ship this!

**Comment by toolmantim** [comment]
@harrietgrace that's now live: https://buildkite.com/docs/how-tos/bitbucket

**Pull request merged**